pkg/app/pipedv1/plugin/kubernetes/provider/manifest.go

```diff
@@ -31,6 +33,23 @@ func (m *Manifest) UnmarshalJSON(data []byte) error {
 	return m.Body.UnmarshalJSON(data)
 }
 
+// MarshalJSON implements the json.Marshaler interface.
+// It marshals the underlying unstructured.Unstructured object into JSON bytes.
+func (m *Manifest) MarshalJSON() ([]byte, error) {
+	return m.Body.MarshalJSON()
+}
+
+// ConvertToStructuredObject converts the manifest into a structured Kubernetes object.
+// The provided interface should be a pointer to a concrete Kubernetes type (e.g. *v1.Pod).
+// It first marshals the manifest to JSON and then unmarshals it into the provided object.
+func (m Manifest) ConvertToStructuredObject(o interface{}) error {
```

Comment:
IMHO, I want to implement without this function, but it's tough to keep compatibility without this.